validation_profiler.gemspec

```diff
@@ -1,46 +1,27 @@
 # coding: utf-8
 lib = File.expand_path('../lib', __FILE__)
 $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
-
-####################
-#### Set GEM Version based on GIT Release TAG when built with Travis
-####################
-version = ''
-if ENV['TRAVIS_TAG'] != nil
-  puts "CI Branch - '#{ENV['TRAVIS_TAG']}'"
-  version = ENV['TRAVIS_TAG']
-end
-
-#if the tag version starts with v (e.g. vx.x.x)
-if version.downcase.match /^v/
-  #trim the v and set the version to x.x.x
-  version = version.dup
-  version.slice!(0)
-elsif ENV['TRAVIS_TAG'] != nil && ENV['TRAVIS_TAG'] != ''
-  version = "0.0.0.#{ENV['TRAVIS_TAG']}"
-else
-  #otherwise it is not a valid release tag so set the version 0.0.0 as it not being released.
-  version = '0.0.0'
-end
+require 'validation_profiler/version'
 
 Gem::Specification.new do |spec|
-  spec.name          = "validation_profiler"
-  spec.version       = version
-  spec.authors       = ["Sage One"]
-  spec.email         = ["vaughan.britton@sage.com"]
+  spec.name          = 'validation_profiler'
+  spec.version       = ValidationProfiler::VERSION
```

Comment:
Why have we reverted back to using the version file rather than using the `vX.X.X` git tag to drive the gem version?

Reply:
It's very difficult to work with the gems when they automatically tag with the git tag locally. This is why we have so much trouble with `build-` gems as they end up being `0.0.0` something and is difficult to get a bundle when the lock is higher without overriding it manually. Setting a `VERSION` makes it very clear what is happening.

Reply:
Ok, how are we going to trigger gem build/push in travis then?

Reply:
Triggering the gem push would work exactly as before, the difference now would be that there is a conscious step/decision to version the gem by updating `ValidationProfiler::VERSION` as part of the release. 

If that's forgotten, the new gem wouldn't be deployed as the version wouldn't have changed.

Reply:
(We're already doing this with gems like https://github.com/Sage/elastic_search_framework, https://github.com/Sage/yaml_normalizer and https://github.com/Sage/fudge )